### ISSUE TITLE
Feature Request: Support atomic move and rename

### ISSUE THREAD

**Issue by Shihira** [althonos/fs.sshfs#33]
Assertion as follows fails. Expect a pass, it is certain that SFTP is powerful enough to support renaming.

```
assert "supports_rename" in remote_fs.getmeta() and remote_fs.getmeta()["supports_rename"]
```

And it costs when moving big files around, which is merely copying the whole file without any optimization. I thought it was paramiko not supporting rename initially but after checking out the [its documentation](https://docs.paramiko.org/en/2.6/api/sftp.html#paramiko.sftp_si.SFTPServerInterface.posix_rename) I found this functionality have already been implemented quite well. Two methods are provided: `rename` and `posix_rename`, both are atomic I suppose.

Thank you very much.

**Comment by althonos** [comment]
Indeed, `fs.sshfs` supports renames, but I wrote the implementation before that meta key was a thing. I'll update that in the next release.

**Comment by althonos** [comment]
Proper implementation available in `v0.12.0`.
